Sources/StreamChat/Models/Attachments/ChatMessageAttachment.swift

```diff
@@ -4,26 +4,51 @@
 
 import Foundation
 
-public struct AttachmentUploadingState: Equatable {
-    public let localFileURL: URL
-    public let state: LocalAttachmentState
-    public let file: AttachmentFile
-}
-
+/// A type representing a chat message attachment.
+/// _ChatMessageAttachment<Payload> is an immutable snapshot of message attachment at the given time.
+///
+/// - Note: _ChatMessageAttachment type is not meant to be used directly. For each specific attachment type
+/// there is a type alias which resolves the generic (`ChatMessageFileAttachment`, `ChatMessageImageAttachment`, etc.).
+/// If you have your own attachment with custom payload consider having a type alias.
 @dynamicMemberLookup
 public struct _ChatMessageAttachment<Payload> {
+    /// The attachment identifier.
     public let id: AttachmentId
+
+    /// The attachment type.
     public let type: AttachmentType
+
+    /// The attachment payload.
     public let payload: Payload
+
+    /// The attachment's file uploading progress.
```

Comment:
-> `The uploading state of the attachment.` ?

Reply:
✅ Updated